### PR TITLE
Implement this expression

### PR DESCRIPTION
**Pull request by sanxiyn**

### CODE REVIEW

**Comment by jasonwilliams** [comment]
@sanxiyn just let me know when you want me to look at this again

**Comment by sanxiyn** [comment]
I rebased. Please have a look.

**Comment by sanxiyn** [comment]
Ping.

**Comment by jasonwilliams** [comment]
@sanxiyn looking, also there's another conflict sorry

File: src/lib/syntax/ast/expr.rs [hunk 1]
```diff
@@ -44,6 +44,7 @@ pub enum ExprDef {
     Block(Vec<Expr>),
     /// Load a reference to a value
     Local(String),
+    This,
```
Comment (jasonwilliams) [inline src/lib/syntax/ast/expr.rs]: A comment above this would be nice

**Comment by Razican** [comment]
Hello, this would need a rebase, and some extra documentation.

**Comment by jasonwilliams** [comment]
superseeded by https://github.com/jasonwilliams/boa/pull/320
(I don't want to delete this PR just yet though)

**Pull request closed without merging**